### ISSUE TITLE
Using localized resources inside a collapsed element causes an infinite layout loop on iPhone 12

### ISSUE THREAD

**Issue by mrlacey** [unoplatform/uno#5960]
I've identified a scenario that can lead to iOS getting stuck in an infinite layout loop, **but only manifests on iPhone 12**.
Although the minimal repo is _very_ trivial, this is based on a real-world app and putting localized content inside an element that may be collapsed is not an unusual thing to do.

## Current behavior

The app never launches as it seems to be stuck in an infinite layout loop.

## Expected behavior

Such a loop should not be possible and scenarios like this should provide debug output to indicate what the problem could be.

## How to reproduce it (as minimally and precisely as possible)

To Reproduce:

1. Create a new app.
2. In `Mainpage.xaml` set the content of the page  to be:
```xml
<Grid>
    <Grid Visibility="Collapsed">
        <StackPanel Padding="20" >
            <TextBox x:Uid="TheTextBox"/>
        </StackPanel>
    </Grid>
</Grid>
```
3. Create the resource (in `Resources.resw`) with **Name** `TheTextBox.Header` and **Value** `Enter something`

4. Run the app on an iOS Simulator for **iPhone 11 - iOS 14.5** and see the app starts.
5. Run the app on an iOS Simulator for **iPhone 12 - iOS 14.5** and see the app never fully launches. It gets stuck on a blank screen, or if you add a launch screen it gets stuck showing that.

## Workaround

At the moment, I've only found the issue arises when the containing element is collapsed, the StackPanel include a value that contains a property that can trigger remeasuring and layout when something changes (Padding in this case) AND when there is content provided by x:Uid and a resw entry.
This is all very specific, but when a complex app that appeared to be working didn't on one group of devices it was extremely time-consuming to work out the cause so a suitable workaround could be employed. Without any indication that there was an infinite layout loop occurring it was very hard to debug. (When I've encountered similar layout loops on Android there is a lot of debug output that makes identifying these easy.)

Knowing the exact factors that cause this, it's possible to create a workaround by structuring the XAML differently or by changing the height rather than the visibility of the containing element.

## Environment

Nuget Package:

- [x] Uno.UI **3.7.3**
- [x] Uno.Core **2.3.0**
- [ ] Uno.SourceGenerationTasks
- [x] Uno.UI.RemoteControl  **3.7.3**
- [ ] Other: 

Nuget Package Version(s):

Affected platform(s):

- [x] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [x] Visual Studio 2019 (version: )
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

Relevant plugins:

- [ ] Resharper (version: )

## Anything else we need to know?

I suspect this may be connected to #5002

**Comment by jeromelaban** [comment]
Thanks for the report! Is this a problem you can only repro on a physical device, or does it fail on the simulator as well ?

**Comment by mrlacey** [comment]
Happens on simulator

**Comment by mrlacey** [comment]
> If you change the Grid for something else like a border, do you still getting the problem?

Yes, it still happens with a `Border` as the collapsed item

**Comment by jeromelaban** [comment]
This looks fixed in latest 3.8-dev builds. Can you revalidate in your app ?

**Comment by mrlacey** [comment]
> This looks fixed in latest 3.8-dev builds. Can you revalidate in your app ?

Yes, this no longer happens with `v3.8.0-dev.130`
